src/clj/game/cards/ice.clj

```diff
@@ -2120,33 +2120,33 @@
 
    "Slot Machine"
    (letfn [(name-builder [card] (str (:title card) " (" (:type card) ")"))
-           (top-3 [runner] (take 3 (:deck runner)))
-           (top-3-names [runner] (map name-builder (top-3 runner)))
-           (top-3-types [runner] (->> (top-3 runner) (map :type) (into #{}) count))]
+           (top-3 [state] (take 3 (get-in @state [:runner :deck])))
+           (top-3-names [state] (map name-builder (top-3 state)))
+           (top-3-types [state] (->> (top-3 state) (map :type) (into #{}) count))]
     {:implementation "Encounter effect is manual"
      :abilities [{:label "Roll them bones"
                   :effect (effect (move :runner (first (:deck runner)) :deck)
                                   (system-msg (str "uses Slot Machine to put the top card of the stack to the bottom,"
                                                    " then reveal the top 3 cards in the stack: "
-                                                   (join ", " (top-3-names runner)))))}]
+                                                   (join ", " (top-3-names state)))))}]
      :subroutines [{:label "Runner loses 3 [Credits]"
                     :msg "force the Runner to lose 3 [Credits]"
                     :effect (effect (lose-credits :runner 3))}
                    {:label "Gain 3 [Credits]"
-                    :effect (req (let [unique-types (top-3-types runner)]
+                    :effect (req (let [unique-types (top-3-types state)]
                                    (when (>= 2 unique-types)
                                      (system-msg state :corp (str "gains 3 [Credits]"))
```

Comment:
Should  move this logging to `:msg` as well.

Reply:
Good call

Reply:
Now that I'm looking at the code, I'd done it this way cuz it felt redundant to do the check twice. I'll change the wording to achieve the same effect as a `:msg`.